modules/apps/headless/headless-admin-user/headless-admin-user-test/src/testIntegration/java/com/liferay/headless/admin/user/resource/v1_0/test/UserGroupResourceTest.java

```diff
@@ -277,6 +335,39 @@ private UserGroup _postUserGroup(UserGroup userGroup) throws Exception {
 		return userGroupResource.postUserGroup(userGroup);
 	}
 
+	private void _testGetUserGroupsPageWithUserBriefs() throws Exception {
+		Page<UserGroup> page = userGroupResource.getUserGroupsPage(
+			null, null, Pagination.of(1, 10), null);
+
+		long totalCount = page.getTotalCount();
+
+		User user1 = UserTestUtil.addUser();
+		User user2 = UserTestUtil.addUser();
+
+		UserGroup userGroup1 = testGetUserUserGroups_addUserGroup(
+			user1.getUserId(), randomUserGroup());
+
+		UserGroup userGroup2 = testGetUserUserGroups_addUserGroup(
+			user2.getUserId(), randomUserGroup());
+
+		page = userGroupResource.getUserGroupsPage(
+			null, null, Pagination.of(1, 10), null);
+
+		Assert.assertEquals(totalCount + 2, page.getTotalCount());
+
+		List<UserGroup> items = (List<UserGroup>)page.getItems();
+
+		assertContains(userGroup1, items);
+		assertContains(userGroup2, items);
+
+		assertValid(page, testGetUserGroupsPage_getExpectedActions());
+
+		_assertContains(TestPropsValues.getUserId(), items, 2);
+
+		_assertContains(user1, items);
+		_assertContains(user2, items);
```

Comment:
this I think can be simplified a lot: in my mind I have this check
* create a user group
* create three users
* add two users to the user group
* call the get with the nestedFields on the user group
* verify that contains the creator node
* verify that contains the userAccountBriefs node with two items where one is the user1 and one is the user2